tck/src/main/java/org/eclipse/microprofile/config/tck/CdiOptionalInjectionTest.java

```diff
@@ -68,6 +73,21 @@ public void testOptionalInjection() {
         Assert.assertEquals(optionalValuesBean.getStringValue().get(), "hello");
 
         Assert.assertFalse(optionalValuesBean.getNotExistingStringProperty().isPresent());
+
+        Assert.assertTrue(optionalValuesBean.getOptionalIntProperty().isPresent());
+        Assert.assertEquals(optionalValuesBean.getOptionalIntProperty().getAsInt(), 1234);
+
+        Assert.assertFalse(optionalValuesBean.getOptionalNotExistingIntProperty().isPresent());
+
+        Assert.assertTrue(optionalValuesBean.getOptionalLongProperty().isPresent());
+        Assert.assertEquals(optionalValuesBean.getOptionalLongProperty().getAsLong(), 1234);
+
+        Assert.assertFalse(optionalValuesBean.getOptionalNotExistingLongProperty().isPresent());
+
+        Assert.assertTrue(optionalValuesBean.getOptionalDoubleProperty().isPresent());
+        Assert.assertEquals(optionalValuesBean.getOptionalDoubleProperty().getAsDouble(), 1234.5);
+
+        Assert.assertFalse(optionalValuesBean.getOptionalNotExistingDoubleProperty().isPresent());
```

Comment:
There should be a non existing check for each type, just to detect the case where an impl did one type correctly but the others incorrectly.